src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,30 +1,71 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
+import React from 'react';
 
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
+type Props = {
+  onQuery: (v: string) => void;
+  onFilter: (v: string) => void;
+  filter: string;
+  query: string;
+};
 
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button data-cy="clearSearchButton" type="button" className="delete" />
-      </span>
-    </p>
-  </form>
-);
+export const TodoFilter: React.FC<Props> = ({
+  onQuery,
+  query,
+  filter,
+  onFilter,
+}) => {
+  const handleChangeQuery = (event: React.ChangeEvent<HTMLInputElement>) => {
+    onQuery(event.target.value);
+  };
+
+  const handleChageFilter = (event: React.ChangeEvent<HTMLSelectElement>) => {
```

Comment:
There's a typo in the function name `handleChageFilter`. It should be `handleChangeFilter` to maintain consistency and avoid confusion.